### PR TITLE
Add miration and global pin for c-blosc2

### PR DESCRIPTION
**Pull request by hmaarrfk**
Larger discussion in: https://github.com/conda-forge/c-blosc2-feedstock/issues/62#issuecomment-2049675391

TLDR:

- Previous run export was to major version
- SO changed in between 2.13 and 2.14

Patching will be made for previous builds in https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/703

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/migrations/cblosc2214.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+migrator_ts: 1712842415
+__migrator:
+  kind: version
+  migration_number: 1
+  bump_number: 1
+
+c_blosc2:
+  - '2.14'
```
Comment (hmaarrfk) [inline recipe/migrations/cblosc2214.yaml:8]:
```suggestion
  - '2.14.4'
```

**Comment by hmaarrfk** [comment]
While the migration should just be to 2.14.X, I want to migrate to 2.14.4 because 2.14.0-throug3 should be considered broken since they use a different SO name.
